[WebCore] CGDisplayStream APIs obsoleted in macOS Sequoia

emw-apple opened the pull request:
#### 79dc79d2431d64c34080a37b799f42842394df77
<pre>
[WebCore] CGDisplayStream APIs obsoleted in macOS Sequoia
<a href="https://rdar.apple.com/132608075">rdar://132608075</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=277189">https://bugs.webkit.org/show_bug.cgi?id=277189</a>

Reviewed by Eric Carlson.

Starting in the public Sequoia SDK, CGDisplayStream APIs are obsoleted.
WebKit has used ScreenCaptureKit to replace CGDisplayStream for years,
but retains fallback logic which calls into the old API. As of this
year, all supported macOS versions have ScreenCaptureKit, so remove
support entirely.

ScreenCaptureKit is not present in the base system, so it is still
possible for ScreenCaptureKitCaptureSource::isAvailable() to be false.
In place of the fallback logic, add ASSERT_NOT_REACHED and bail out.

* Source/WTF/wtf/PlatformHave.h: Remove the version check for
  HAVE(SCREEN_CAPTURE_KIT); all supported releases have it.

Source file deletions:
* Source/WebCore/SmartPointerExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/platform/mediastream/cocoa/DisplayCaptureSourceCocoa.cpp:
(WebCore::DisplayCaptureSourceCocoa::create): Remove fallback.

* Source/WebCore/platform/mediastream/mac/CGDisplayStreamCaptureSource.cpp: Removed.
* Source/WebCore/platform/mediastream/mac/CGDisplayStreamCaptureSource.h: Removed.
* Source/WebCore/platform/mediastream/mac/CGDisplayStreamScreenCaptureSource.h: Removed.
* Source/WebCore/platform/mediastream/mac/CGDisplayStreamScreenCaptureSource.mm: Removed.
* Source/WebCore/platform/mediastream/mac/DisplayCaptureManagerCocoa.cpp:
(WebCore::DisplayCaptureManagerCocoa::screenCaptureDeviceWithPersistentID): Remove fallback.

* Source/WebKit/UIProcess/mac/DisplayCaptureSessionManager.mm:
(WebKit::DisplayCaptureSessionManager::promptForGetDisplayMedia): Remove
unreachable fallback logic.

Canonical link: <a href="https://commits.webkit.org/281529@main">https://commits.webkit.org/281529@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4cb66d138cce0208f350b57c8db8b91a2b518993

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60151 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39499 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12703 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64069 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10681 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62281 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47171 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10913 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48725 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7457 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62182 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36827 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52110 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29567 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33532 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9343 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9601 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/53245 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55450 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9621 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65801 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/59396 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4081 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9503 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56078 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4099 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52087 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56233 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13355 "") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3398 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/81154 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35312 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14099 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36394 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37482 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36139 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebCore/platform/mediastream/cocoa/DisplayCaptureSourceCocoa.cpp:
```diff
@@ -67,12 +66,13 @@ CaptureSourceOrError DisplayCaptureSourceCocoa::create(const CaptureDevice& devi
             return create(ScreenCaptureKitCaptureSource::create(device, constraints), device, WTFMove(hashSalts), constraints, pageIdentifier);
-#endif
-        return create(CGDisplayStreamScreenCaptureSource::create(device.persistentId()), device, WTFMove(hashSalts), constraints, pageIdentifier);
+        ASSERT_NOT_REACHED();
```
Seems like this falls through to `case CaptureDevice::DeviceType::Window` in release builds?

eric-carlson replied:
Indeed, please add a `return { }`

emw-apple committed:
[WebCore] CGDisplayStream APIs obsoleted in macOS Sequoia
rdar://132608075
https://bugs.webkit.org/show_bug.cgi?id=277189

Reviewed by Eric Carlson.

Starting in the public Sequoia SDK, CGDisplayStream APIs are obsoleted.
WebKit has used ScreenCaptureKit to replace CGDisplayStream for years,
but retains fallback logic which calls into the old API. As of this
year, all supported macOS versions have ScreenCaptureKit, so remove
support entirely.

ScreenCaptureKit is not present in the base system, so it is still
possible for ScreenCaptureKitCaptureSource::isAvailable() to be false.
In place of the fallback logic, add ASSERT_NOT_REACHED and bail out.

* Source/WTF/wtf/PlatformHave.h: Remove the version check for
  HAVE(SCREEN_CAPTURE_KIT); all supported releases have it.

Source file deletions:
* Source/WebCore/SmartPointerExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/platform/mediastream/cocoa/DisplayCaptureSourceCocoa.cpp:
(WebCore::DisplayCaptureSourceCocoa::create): Remove fallback.

* Source/WebCore/platform/mediastream/mac/CGDisplayStreamCaptureSource.cpp: Removed.
* Source/WebCore/platform/mediastream/mac/CGDisplayStreamCaptureSource.h: Removed.
* Source/WebCore/platform/mediastream/mac/CGDisplayStreamScreenCaptureSource.h: Removed.
* Source/WebCore/platform/mediastream/mac/CGDisplayStreamScreenCaptureSource.mm: Removed.
* Source/WebCore/platform/mediastream/mac/DisplayCaptureManagerCocoa.cpp:
(WebCore::DisplayCaptureManagerCocoa::screenCaptureDeviceWithPersistentID): Remove fallback.

* Source/WebKit/UIProcess/mac/DisplayCaptureSessionManager.mm:
(WebKit::DisplayCaptureSessionManager::promptForGetDisplayMedia): Remove
unreachable fallback logic.

Canonical link: https://commits.webkit.org/281529@main
